### ISSUE TITLE
Question/feature request: possibility to exclude functions annotated with given annotation (e.g. Composables) from coverage calculation

### ISSUE THREAD

**Issue by plnice** [Kotlin/kotlinx-kover#121]
One of the biggest struggles with code coverage in my Android project is that we have a very little code coverage reported by the JaCoCo due to the fact that the project uses Jetpack Compose. We calculate unit tests coverage, however, Composables should be tested using UI tests (https://developer.android.com/jetpack/compose/testing). We'd like to exclude `@Composable`-annotated Kotlin functions from unit tests coverage calculation, however, there's no way to do this with JaCoCo (https://github.com/jacoco/jacoco/issues/1208, https://github.com/jacoco/jacoco/pull/1259). If kover would support to exclude such code, that would be a game changer for us and definitely we would switch from JaCoCo to kover, once it's stable.

**Comment by stantronic** [comment]
Related feature idea - provide an annotation like @KoverExclude that can be manually added to classes and functions to exclude from coverage. This would be preferable in my view to long lists of exclusions in gradle files which lack the context of what those classes are.

**Comment by qwwdfsad** [comment]
I second the idea -- it would be nice to both have `@Generated` package-agnostic annotation to exclude autogenerated code as well as user-provided one, such as `@Composable`

**Comment by rbrauwers** [comment]
> Related feature idea - provide an annotation like @KoverExclude that can be manually added to classes and functions to exclude from coverage. This would be preferable in my view to long lists of exclusions in gradle files which lack the context of what those classes are.

@stantronic suggestion is great. As one use case, it would make possible remove Compose @Preview(s) from test coverage.

**Comment by headsvk** [comment]
I tried using `@Generated` on previews with Jacoco and the problem was that it doesn't apply to lambdas and it got verbose tagging all lambdas. Would be nice if Kover could handle it automatically.

```
@ExcludedFromGeneratedReports
@Preview(backgroundColor = 0xFFFFFFFF, showBackground = true)
@Composable
private fun HomePagePreview() {
    Theme @ExcludedFromGeneratedReports {
        HomePage(
            title = "Hello World",
            onClose = @ExcludedFromGeneratedReports {},
        )
    }
}
```

Since https://github.com/jacoco/jacoco/issues/1208 was closed as won't fix, I would gladly switch to the IntelliJ agent if it provided better support for Compose.

**Comment by samuelneff** [comment]
Huge `+1000` for this. When I worked in JS world we would annotate individual lines or blocks that could not be covered and then set the code coverage requirement to 100%. It was far more powerful than an arbitrary number like 80% where we would not know what the 20% uncovered actually was.

**Comment by shanshin** [comment]
Implemented in `0.6.1`
